src/gentropy/datasource/finngen_ukb_meta/study_index.py

```diff
@@ -0,0 +1,62 @@
+"""Study Index for Finngen data source."""
+from __future__ import annotations
+
+import pyspark.sql.functions as f
+from pyspark.sql import SparkSession
+
+from gentropy.dataset.study_index import StudyIndex
+
+
+class FinngenUkbMetaStudyIndex(StudyIndex):
+    """Study index dataset from FinnGen UKB meta-analysis."""
+
+    @classmethod
+    def from_source(
+        cls: type[FinngenUkbMetaStudyIndex],
+        spark: SparkSession,
+        raw_study_index_path: str,
+    ) -> StudyIndex:
+        """This function ingests study level metadata from FinnGen UKB meta-analysis.
+
+        Args:
+            spark (SparkSession): Spark session object.
+            raw_study_index_path (str): Raw study index path.
```

Comment:
Mayby ensure that the input path is going to be tsv file? It would be usefull to add this info to the docstring or even name it `from_tsv`

Reply:
God point, I removed it to `raw_study_index_path_from_tsv`